### PR TITLE
첫번째 커밋

### PR DESCRIPTION
**Pull request by leekeunha**
1.프로젝트 폴더 구조 형성
2.product,category,httpclient
3.유닛테스트

### CODE REVIEW

File: src/app/httpClient.ts [hunk 1]
```diff
@@ -6,17 +6,39 @@ import * as categoriesData from './categories.json';
...
+    filterByProductId(productList: Product[], productId: number): Product {
+        return productList.find(p => p.product_id == productId);
+    }
+
+    filterByParentCategoryId(productList: Product[], parentCategoryId: number): Product[] {
+        return productList.filter(p => p.parent_category_id == parentCategoryId);
+    }
+
+    filterByName(productList: Product[], productName: string): Product {
+        return productList.find(p => p.product_name == productName);
```
Comment (tomlee-cmkr) [inline src/app/httpClient.ts:34]: change the logic `p.product_name` contains `productName`

File: src/app/tests/httpClient.spec.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { TestBed, ComponentFixture } from "@angular/core/testing";
+import { Product } from "../models/product.model";
+import { Category } from "../models/category.model";
+import { ParentCategory } from "../models/parentCategory.model";
+import { HttpClient } from '../httpClient';
+
+describe("Jasmine Test Environment", () => {
```
Comment (tomlee-cmkr) [inline src/app/tests/httpClient.spec.ts:7]: `Jasmine Test Environment` doesn't describe what is this test for.
change the description as what is the purpose of this test

File: src/app/tests/httpClient.spec.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    it("test getProducts function", () => {
```
Comment (tomlee-cmkr) [inline src/app/tests/httpClient.spec.ts:8]: let's apply description as `it ('should ..` style
for example, it('should return a list of products JSON objects`)

ref: https://howtodoinjava.com/javascript/jasmine-unit-testing-tutorial/

File: src/app/tests/httpClient.spec.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        let httpClient: HttpClient = new HttpClient();
+        expect(httpClient.getProducts().length).toBe(38);
+    });
```
Comment (tomlee-cmkr) [inline src/app/tests/httpClient.spec.ts:11]: always check at least one object to be expected. Do not only check the length of the array

File: src/app/tests/httpClient.spec.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    it("test getParentCategories function", () => {
+        let httpClient: HttpClient = new HttpClient();
+        expect(httpClient.getParentCategories().length).toBe(5);
+    });
+    it("test getProductById function", () => {
+        let httpClient: HttpClient = new HttpClient();
+        expect(httpClient.getProductById(1).product_id).toBe(1);
```
Comment (tomlee-cmkr) [inline src/app/tests/httpClient.spec.ts:18]: check every property of the returned object

**Review by tomlee-cmkr — CHANGES_REQUESTED** [review]
1. create `client` folder and move client.ts into `client folder.
2. create `fixture` folder under `client` folder and move json files into `fixture` folder

File: src/app/tests/httpClient.spec.ts (hunk 5 above)
Comment (leekeunha) [reply src/app/tests/httpClient.spec.ts:18]: 수정완료

File: src/app/httpClient.ts (hunk 1 above)
Comment (leekeunha) [reply src/app/httpClient.ts:34]: 수정완료

**Pull request merged**